redgenes/gene_annotations.py

```diff
@@ -227,11 +232,33 @@ def annotation_pipeline(df, tmpdir, kofamscan_profile, kofamscan_kolist):
             args_md_info = entity_id + args_md_info
             TRN.add(sql_md_info, args_md_info)
 
+            prodigal_version = cds_df["source"].tolist()[0]
+            sql_extract_prodigal_run_info = """
+                SELECT run_id FROM run_info
+                WHERE software = 'prodigal' and version = ? and commands = ?"""
+            TRN.add(
+                sql_extract_prodigal_run_info, [prodigal_version, prodigal_commands]
+            )
+            prodigal_run_id = TRN.execute_fetchflatten()
+
+            if not prodigal_run_id:
```

Comment:
oh I see, this is essentially an "if exists" type thing.

One thing that we might want to do, not in this pull request but perhaps as a refactor, is setup data access objects with the general crud operations. eg something like:

```python
class RunInfo:
    def __init__(self):
        # unsure if stuff needs t obe done here for this object
    def create(self, software, version, command):
        # i think we wnat this to be idempotent, so it always returns the `run_id`
        # so select the id if it exists, or insert the entry and return the new id if it doesnt
    def update(self, *args, **kwargs):
        # i think this isnt' valid for run info so we would raise
        raise NotImplementedError("update is not supported")
    def delete(self, *args, **kwargs):
        # same here
        raise NotImplementedError("delete is not supported")
    def select(self, software, version, command):
        # get a run_id or None if one doesn't exist
```

A benefit to this type of structure is it abstracts out the sql being run, so as a developer, the focus is on very specific terms like "create", or "select"